### PR TITLE
up torch version compatibility

### PR DESCRIPTION
**Pull request by JRosenkranz**
Switched torch version to ~=2.2

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Used to install pinned dependencies
 # Useful for dev/test jobs caches
 # Must be kept in sync with setup.py
-torch == 2.2.0 # This is what is installed in CI today
+torch ~= 2.2 # This is what is installed in CI today
```
Comment (nairbv) [inline requirements.txt:4]: I think we also have an `install_requires=["torch >= 2.1"],` in setup.py that we should probably also make consistent

Comment (JRosenkranz) [reply requirements.txt:4]: I am not seeing that here:

```
install_requires=get_requirements()
```

Comment (nairbv) [reply requirements.txt:4]: ah I see, I was looking in the wrong repo. We may want to update the others to be consistent after this:

https://github.com/foundation-model-stack/fms-fsdp/blob/main/setup.py#L12

and

https://github.com/foundation-model-stack/foundation-model-stack/blob/main/setup.py#L12

**Review by nairbv — APPROVED** [review]

File: requirements.txt (hunk 1 above)
Comment (JRosenkranz) [reply requirements.txt:4]: Will open PR there as well shortly

**Pull request merged**